src/devices/device_models/common/cost_functions.jl

```diff
@@ -129,15 +129,51 @@ function _pwlparamcheck(cost_)
     return flag
 end
 
+function _gen_cost(canonical_model::CanonicalModel,
+                    variable::JV,
+                    cost_component::Vector{NTuple{2, Float64}}) where {JV<:JuMP.AbstractVariableRef}
+    
+    if !_pwlparamcheck(cost_component) 
+        @warn("Data provide is not suitable for linear implementation of PWL cost, Using SOS-2 implementation") ;
```

Comment:
The error message should be a bit more descriptive and also has a typo. how about 
```The cost function provided for device $() is not compatible with a linear PWL cost function. An SOS-2 formulation will be added to the model. This will result in additional binary variables added to the model.```